app/src/org/commcare/dalvik/activities/EntitySelectActivity.java

```diff
@@ -502,6 +506,8 @@ protected void onPause() {
         if (adapter != null) {
             adapter.unregisterDataSetObserver(mListStateObserver);
         }
+
+        hereFunctionHandler.stopLocationUpdates();

```

Comment:
The entities are reloaded in `onResume`, which calls, `eval`, which then calls `HereFunctionHandler.requestLocationUpdates`. Doesn't that restart location acquisition?